client/transactions/list/index.tsx

```diff
@@ -383,20 +388,100 @@ export const TransactionsList = (
 
 	const downloadable = !! rows.length;
 
-	const onDownload = () => {
+	const onDownload = async () => {
+		setIsDownloading( true );
+
 		// We destructure page and path to get the right params.
 		// eslint-disable-next-line @typescript-eslint/no-unused-vars
 		const { page, path, ...params } = getQuery();
+		const downloadType = totalRows > rows.length ? 'endpoint' : 'browser';
+
+		if ( 'endpoint' === downloadType ) {
+			const {
+				date_after: dateAfter,
+				date_before: dateBefore,
+				date_between: dateBetween,
+				match,
+				search,
+				type_is: typeIs,
+				type_is_not: typeIsNot,
+			} = params;
+
+			const isFiltered =
+				!! dateAfter ||
+				!! dateBefore ||
+				!! dateBetween ||
+				!! search ||
+				!! typeIs ||
+				!! typeIsNot;
+
+			const confirmThreshold = 1000;
```

Comment:
I arbitrarily set this to be 1000, but am happy to bump this up to something higher. I'm not really sure what the typical WCPay merchant might have here.

Reply:
> why do we need to set a threshold on the client-side

Good question! The threshold is to give merchants a chance to filter prior to kicking off the export. For example, a merchant might click download believing they don't have so many transactions. If their export size is larger than the threshold the merchant can rethink this decision and opt to add some filters for a potentially smaller export.

>  IMO the server should be able to handle, 1k - 10k export easily, wdyt?

Yeah. That was my initial hunch. 1K was a bit too conservative. I've upped this to 10k in https://github.com/Automattic/woocommerce-payments/pull/3477/commits/aa7647f2a055b23da9341ba3413e8d4b059ae13f

Reply:
Sorry, I realized I forgot to address this question directly: 

> what if the client has more than 1000 transactions?

If the merchant has over `confirmThreshold` transactions, they can either confirm and get all of these in their export, OR cancel and select some filters to reduce the size of their export.